module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -690,12 +690,78 @@ public function getMyTransactions($patron)
                 'id' => $this->getBibId($trans->item->instanceId),
                 'item_id' => $trans->item->id,
                 'barcode' => $trans->item->barcode,
+                'renew' => $trans->renewalCount ?? 0,
+                'renewable' => true,
                 'title' => $trans->item->title,
             ];
         }
         return $transactions;
     }
 
+    /**
+     * Get FOLIO loan IDs for use in renewMyItems.
+     *
+     * @param array $transaction An single transaction
+     * array from getMyTransactions
+     *
+     * @return string The FOLIO loan ID for this loan
+     */
+    public function getRenewDetails($transaction)
+    {
+        $loanId = $transaction['item_id'];
+        return $loanId;
+    }
+
+    /**
+     * Attempt to renew a list of items for a given patron.
+     *
+     * @param array $renewDetails An associative array with
+     * patron and details
+     *
+     * @return array $renewResult result of attempt to renew loans
+     */
+    public function renewMyItems($renewDetails)
+    {
+        $userId = $renewDetails['patron']['id'];
+        $renewalResults = [];
+
+        foreach ($renewDetails['details'] as $loanId) {
+            $renewal = [];
+            $requestbody = [
+                'itemId' => $renewDetails['details'][0],
```

Comment:
Should this be $loanId rather than $renewDetails['details'][0]? Otherwise, it looks like if a user renews N books, you will just renew the first item N times.

Reply:
THANK YOU!